cmd/main_test.go

```diff
@@ -0,0 +1,490 @@
+		NotAfter:              time.Now().Add(24 * time.Hour),
+		BasicConstraintsValid: true,
+	}
+
+	if len(uris) > 0 {
+		template.URIs = make([]*url.URL, len(uris))
+		for i, uri := range uris {
+			template.URIs[i] = parseURL(uri)
+		}
+	}
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatalf("Failed to generate private key: %v", err)
+	}
+
+	certDER, err := x509.CreateCertificate(rand.Reader, template, template, &privateKey.PublicKey, privateKey)
+	if err != nil {
+		t.Fatalf("Failed to create certificate: %v", err)
+	}
+
+	cert, err := x509.ParseCertificate(certDER)
+	if err != nil {
+		t.Fatalf("Failed to parse certificate: %v", err)
+	}
+
+	return cert, privateKey
+}
+
+// createTestSVID creates an SVID for testing.
+func createTestSVID(t *testing.T, td, path string) *x509svid.SVID {
+	cert, _ := createTestCertificate(t, "test."+td, "spiffe://"+td+path)
+	return &x509svid.SVID{
+		ID:           spiffeid.RequireFromPath(spiffeid.RequireTrustDomainFromString(td), path),
+		Certificates: []*x509.Certificate{cert},
+	}
+}
+
+func createBundleSet(t *testing.T) *x509bundle.Set {
+	td, err := spiffeid.TrustDomainFromString("domain.test")
+	if err != nil {
+		t.Fatalf("Failed to create trust domain: %v", err)
+	}
+	bundle := x509bundle.New(td)
+	return x509bundle.NewSet(bundle)
+}
+
+func createEmptyBundleSet(t *testing.T) *x509bundle.Set {
+	return x509bundle.NewSet()
+}
+
+func createBundleSetWithoutAuthority(t *testing.T) *x509bundle.Set {
+	td, err := spiffeid.TrustDomainFromString("domain.test")
+	if err != nil {
+		t.Fatalf("Failed to create trust domain: %v", err)
+	}
+	bundle := x509bundle.New(td)
+	set := x509bundle.NewSet(bundle)
+	return set
+}
+
+func createBundleSetWithAuthority(t *testing.T) *x509bundle.Set {
+	td, err := spiffeid.TrustDomainFromString("domain.test")
+	if err != nil {
+		t.Fatalf("Failed to create trust domain: %v", err)
+	}
+
+	template := &x509.Certificate{
+		Subject: pkix.Name{
+			CommonName: "cofide.io",
+		},
+		Issuer: pkix.Name{
+			CommonName: "cofide.io",
+		},
+		SerialNumber:          big.NewInt(1),
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(24 * time.Hour),
+		IsCA:                  true,
+		BasicConstraintsValid: true,
+		KeyUsage:              x509.KeyUsageCertSign,
+		SignatureAlgorithm:    x509.SHA256WithRSA,
+		URIs:                  []*url.URL{parseURL("spiffe://domain.test")},
+	}
+
+	bundle := x509bundle.New(td)
+	bundle.AddX509Authority(template)
+	set := x509bundle.NewSet(bundle)
+	return set
+}
+
+func createBundleSetWithCert(t *testing.T, cert *x509.Certificate) *x509bundle.Set {
+	td, err := spiffeid.TrustDomainFromString("domain.test")
+	if err != nil {
+		t.Fatalf("Failed to create trust domain: %v", err)
+	}
+	bundle := x509bundle.New(td)
+	bundle.AddX509Authority(cert)
+	return x509bundle.NewSet(bundle)
+}
+
+// captureOutput captures the output of a function and returns it as a string.
```

Comment:
No need to change, but another way to do this is to make the code under test more amenable to testing, i.e. decouple from IO. e.g. https://dev.to/lucassha/unit-testing-stdout-in-go-1jd